metrics/metrics/src/main/java/io/helidon/metrics/HelidonConcurrentGauge.java

```diff
@@ -183,17 +210,31 @@ public synchronized void dec() {
         }
 
         public synchronized void updateState() {
```

Comment:
I do not quite get the point of `synchronized` here and in both `inc` and `dec`.

`synchronized` `inc` makes sense only if all accesses to `count` and `currentMax` are also `synchronized`, including reads, in which case we don't need them to be `AtomicLong`. But I presume they are `AtomicLong` to enable atomic updates that are cheaper than `synchronized`. In which case we don't need `synchronized`.

Atomic updates should be like so:

```
public void inc() {
   long count = count.incrementAndGet();
   if (count > currentMax.get()) currentMax.getAndUpdate(v -> count > v ? count: v);
}
```

`updateState` should probably be done only by the minute-updating routine.

And to check the intention: the `currentMin` and `currentMax` don't get reset after every minute intentionally?